src/core/model_concrete.jl

```diff
@@ -0,0 +1,257 @@
+
+
+The most standard concrete implementation of an [`AgentBasedModel`](@ref),
+as well as the default version of the generic [`AgentBasedModel`](@ref) constructor.
+`StandardABM` stores agents in a dictionary mapping unique `Int` IDs to agents.
+See also [`UnkillableABM`](@ref), [`FixedMassABM`](@ref).
+"""
+StandardABM(args...; kwargs...) = SingleContainerABM(args...; kwargs..., container=Dict{Int})
+
+"""
+    UnkillableABM(AgentType [, space]; properties, kwargs...) → model
+
+Similar to [`StandardABM`](@ref), but agents cannot be removed, only added.
+This allows storing agents more efficiently in a standard Julia `Vector` (as opposed to
+the `Dict` used by [`StandardABM`](@ref), yielding faster retrieval and iteration over agents.
+
+It is mandatory that the agent ID is exactly the same as the agent insertion
+order (i.e., the 5th agent added to the model must have ID 5). If not,
+an error will be thrown by [`add_agent!`](@ref).
+"""
+UnkillableABM(args...; kwargs...) = SingleContainerABM(args...; kwargs..., container=Vector)
+
+"""
+    FixedMassABM(agent_vector [, space]; properties, kwargs...) → model
+
+Similar to [`UnkillableABM`](@ref), but agents cannot be removed nor added.
+Hence, all agents in the model must be provided in advance as a vector.
+This allows storing agents into a `SizedVector`, a special vector with statically typed
+size which is the same as the size of the input `agent_vector`.
+This version of agent based model offers better performance than [`UnkillableABM`](@ref)
+if the number of agents is important and used often in the simulation.
+
+It is mandatory that the agent ID is exactly the same as its position
+in the given `agent_vector`.
+"""
+function FixedMassABM(
+    agents::AbstractVector{A},
+    space::S = nothing;
+    scheduler::F = Schedulers.fastest,
+    properties::P = nothing,
+    rng::R = Random.default_rng(),
+    warn = true
+) where {A<:AbstractAgent, S<:SpaceType,F,P,R<:AbstractRNG}
+    C = SizedVector{length(agents), A}
+    fixed_agents = C(agents)
+    # Validate that agent ID is the same as its order in the vector.
+    for (i, a) in enumerate(agents)
+        i ≠ a.id && throw(ArgumentError("$(i)-th agent had ID $(a.id) instead of $i."))
+    end
+    agent_validator(A, space, warn)
+    return SingleContainerABM{S,A,C,F,P,R}(fixed_agents, space, scheduler, properties, rng, Ref(0))
+end
+
+#######################################################################################
+# %% Model accessing api
+#######################################################################################
+nextid(model::SingleContainerABM{<:SpaceType,A,Dict{Int, A}}) where {A} = getfield(model, :maxid)[] + 1
+nextid(model::SingleContainerABM{<:SpaceType,A,Vector{A}}) where {A} = nagents(model) + 1
+nextid(::SingleContainerABM{<:SpaceType,A,SizedVector}) where {A} = error("There is no `nextid` in a `FixedMassABM`. Most likely an internal error.")
```

Comment:
```suggestion
nextid(::SingleContainerABM{<:SpaceType,A,<:SizedVector}) where {A} = error("There is no `nextid` in a `FixedMassABM`. Most likely an internal error.")
```

This won't work without the `<:`:
```
julia> f(::Vector{Integer}) = 3
f (generic function with 1 method)

julia> f(x) = 2
f (generic function with 2 methods)

julia> f([])
2
julia> f([1])
2

julia> f(::Vector{<:Integer}) = 3
f (generic function with 3 methods)

julia> f([1])
3
```